### PR TITLE
add server spawn delay

### PR DESCRIPTION
**Pull request by shaneknapp**
see https://github.com/berkeley-dsep-infra/datahub/issues/4237 and https://github.com/berkeley-dsep-infra/datahub/issues/4009#issuecomment-1423335899

### CODE REVIEW

**Review by felder — COMMENTED** [review]
Were you able to test this in staging via hubploy first?

File: hub/values.yaml [hunk 1]
```diff
@@ -328,6 +337,10 @@ jupyterhub:
               self.mem_guarantee = override['mem_guarantee']
 
           async def start(self):
+            # delay if testing, see https://github.com/berkeley-dsep-infra/datahub/issues/4237
+            if self.spawn_launcher_delay:
```
Comment (ryanlovett) [inline hub/values.yaml:341]: Will this always be True if the default is `180` ? Should the default be `0` for that reason?

I might not know the behavior of traitlets well enough -- maybe if the admin doesn't actually set the var, then the `if self.spawn_launcher_delay` will fail. Based on the [docs](https://traitlets.readthedocs.io/en/stable/config.html#python-configuration-files), I think the default value will be used.

So basically I think the default value should be 0, and the `if` should check for non-zero.

Comment (shaneknapp) [reply hub/values.yaml:341]: default will indeed be 0.  i set it to 180 for testing purposes.

**Comment by shaneknapp** [comment]
> Were you able to test this in staging via hubploy first?

yep, only a11y-staging has this new "feature" deployed.

File: hub/values.yaml (hunk 1 above)
Comment (shaneknapp) [reply hub/values.yaml:341]: and i pushed a new commit setting this back to 0

**Comment by shaneknapp** [comment]
ok, this works and has already shown itself to be useful!  will be merging to staging now.
